docs/RELEASE.md

```diff
@@ -0,0 +1,77 @@
+# Release Process
+
+## Overview
+
+The release process consists of three phases: versioning, building and publishing
+
+Versioning involves maintaining the following files:
+
+- Makefile - the Makefile contains a `VERSION` variable that defines the version of the project.
+- secrets-store-csi-driver.yaml - the Linux driver deployment yaml that contains the latest release tag image of the project.
+- secrets-store-csi-driver-windows.yaml - the Windows driver deployment yaml that contains the latest release tag image of the project.
+- `deploy/` dir that contains all the secrets-store-csi-driver resources to be deployed to the cluster including the latest release tag image of the project.
+
+The steps below explain how to update these files. In addition, the repository should be tagged with the semantic version identifying the release.
+
+Building involves obtaining a copy of the repository and triggering an automatic build as part of the [prow job](https://testgrid.k8s.io/sig-auth-secrets-store-csi-driver#secrets-store-csi-driver-push-image).
+
+Publishing involves creating a release tag and creating a new Release on GitHub.
+
+## Versioning
+
+1. Update the version to the semantic version of the new release similar to [this](https://github.com/kubernetes-sigs/secrets-store-csi-driver/pull/251)
+1. Commit the changes and push to remote repository to create a pull request
+
+    ```
+    git checkout -b bump-version-<NEW VERSION>
+    git commit -m "Bump versions for <NEW VERSION"
+    git push <YOUR FORK>
+    ```
+   
+ 1. Once the PR is merged to master, the [prow job](https://testgrid.k8s.io/sig-auth-secrets-store-csi-driver#secrets-store-csi-driver-push-image) is triggered to build and push the new version to staging repo (`gcr.io/k8s-staging-csi-secrets-store/driver`)
+ 1. Once the prow job completes, follow the [instructions](https://github.com/kubernetes/k8s.io/tree/master/k8s.gcr.io#image-promoter) to promote the image to production repo
+    - Run generate script to append the new image to promoter manifest
+    ```bash
+    k8s.gcr.io/images/k8s-staging-csi-secrets-store/generate.sh > k8s.gcr.io/images/k8s-staging-csi-secrets-store/images.yaml
+    ```
+    - Preview the changes
+    ```bash
+    git diff
+    ```
+    - Commit the changes and push to remote repository to create a pull request
+  1. Once the image promoter PR is merged, the image will be promoted to prod repo (`us.gcr.io/k8s-artifacts-prod/csi-secrets-store/driver`)
+  
+## Building and releasing
+
+1. Execute the release-patch target to generate patch. Give the semantic version of the release:
+    ```
+   make release-manifests NEWVERSION=v0.0.12
```

Comment:
If `make release-manifest` is only used when we cut a release, then you can have `promote-staging-manifest` call it so we only need to run a single command.